### DISCUSSION TITLE
How can search and install become unsync on client side for custom registry

### DISCUSSION THREAD

**Discussion by Thieum** [microsoft/vcpkg#18852] [Q&A] [answered]
I certainly made a mistake during the process of adding a new package in my registry. 

I am able to find the package, but I cannot install it.

![image](https://user-images.githubusercontent.com/944894/124783597-30247380-df13-11eb-8fab-dfd0672966b9.png)

There is the git-tree that works for the search:

![image](https://user-images.githubusercontent.com/944894/124784365-db352d00-df13-11eb-9a4b-e02560ceb3e0.png)

But when installing, another git-tree is pulled, which is older and not working:

![image](https://user-images.githubusercontent.com/944894/124784435-edaf6680-df13-11eb-9537-6358d9ae6ee2.png)

My vcpkg-configuration.json is written this way:

```
{
    "default-registry": {
        "kind": "git",
        "repository": "http://10.1.102.168/Bonobo.Git.Server/vcpkg-registry.git",
        "baseline": "df99aca6cb95547e0b8af076f04b494fae96fff7"
    },
    "registries": [
      {
        "kind": "git",
        "repository": "http://10.1.102.168/Bonobo.Git.Server/vcpkg-registry.git",
        "baseline": "df99aca6cb95547e0b8af076f04b494fae96fff7",
        "packages": [ "csvhelper", "stringhelper" ]
      },
      {
        "kind": "builtin",
        "baseline": "6cdc26c884ac1c7c4cf2d480b364b773765622a3",
        "packages": [ "gtest", "paho-mqtt" ]
      }
    ]
  }
```
  
  I would expect search and install to work the same way, at least to pull the resources. I wonder what I'm missing here?

Edit:

using --debug I found that the git commands were using two different commit ids to grab the files - the first one for baseline,json is the good one, the second one for versions is too old: 
![image](https://user-images.githubusercontent.com/944894/124938912-b0f87380-dfd6-11eb-81d7-e9f6475ec86e.png)

Where the second one comes from? What could be the source of this old commit id beside from vcpkg-configuration.json?

Edit:
I managed to bypass this issue by:
 - creating a newer version of the first package I created;
 - installing this newer version locally;
 - installing the other package

It still feels like a hack, and not really sustainable if I have to create new port for every packages everytime I create a new package.

**Comment by mathisloge** [comment, marked as answer]
So i don't really know if this is the fix for your problem, but you also have the default-registry under the registries. All packages are pulled from the default registry by default. A registry that is under registries adds ports or overwrites the default registry.

**Reply by Thieum** [reply]
It seems better this way indeed. At least, I managed to push an new version of my library, without updating the other one. Thanks!
